content/en/blog/_posts/2021-XX-XX-writing-a-controller-for-pod-labels.md

```diff
@@ -0,0 +1,473 @@
+---
+layout: blog
+title: "Writing a Controller for Pod Labels"
+date: 2021-03-XX
+slug: writing-a-controller-for-pod-labels
+---
+
+**Authors**: Arthur Busser, Padok
+
+[Operators][what-is-an-operator] are proving to be an excellent solution to
+running stateful distributed applications in Kubernetes. Open Source tools like
+the [Operator SDK][operator-sdk] provide ways to build reliable and maintainable
+operators, and extending Kubernetes to implement custom scheduling needs is now
+easier than ever.
+
+Kubernetes operators run complex software inside your cluster. The open-source
+community has already built [many operators][operatorhub] for distributed
+applications like Prometheus, ElasticSearch, or ArgoCD. Even outside of
+open-source, operators can help to bring new functionality to your Kubernetes
+cluster.
+
+An Operator is a set of [custom resources][custom-resource-definitions] and a
+set of [controllers][controllers] that, when combined, run complex software
+inside your Kubernetes cluster. A controller watches for changes to specific
+resources in the Kubernetes API and reacts by creating, updating, or deleting
+resources.
+
+The Operator SDK is best suited for building fully fledged operators.
+Nonetheless, you can use it to write a single controller. This post will walk
+you through writing a Kubernetes controller in Go that will add a `pod-name`
+label to Pods that have a specific annotation.
+
+## Why do we need a controller for this?
+
+I recently worked on a project where we needed to create a Service that routed
```

Comment:
This is really helpful. Hearing you ("I" and "we") relate your experiences and that of your team or organization gives weight to the use story you're telling.

Reply:
Do you think I should add more or is this enough?

Reply:
This is good. I don't want to add more review cycles.